cmake/utilities.cmake

```diff
@@ -0,0 +1,489 @@
+    target_compile_definitions(global_settings INTERFACE
+      WIN32=1
+      WINDOWS=1
+      OSQUERY_WINDOWS=1
+      OSQUERY_BUILD_PLATFORM=windows
+      OSQUERY_BUILD_DISTRO=10
+      BOOST_ALL_NO_LIB
+      BOOST_ALL_STATIC_LINK
+      _WIN32_WINNT=_WIN32_WINNT_WIN7
+      NTDDI_VERSION=NTDDI_WIN7
+    )
+  else()
+    message(FATAL_ERROR "This platform is not yet supported")
+  endif()
+
+  add_library(c_settings INTERFACE)
+  add_library(cxx_settings INTERFACE)
+
+  # C++ settings
+  if (DEFINED PLATFORM_WINDOWS)
+    target_compile_options(cxx_settings INTERFACE
+      /MT
+      /EHs
+      /W3
+      /guard:cf
+      /bigobj
+      /Zc:inline-
+    )
+  else()
+    target_compile_options(cxx_settings INTERFACE
+      -Qunused-arguments
+      -Wno-shadow-field
+      -Wall
+      -Wextra
+      -Wno-unused-local-typedef
+      -Wno-deprecated-register
+      -Wno-unknown-warning-option
+      -Wstrict-aliasing
+      -Wno-missing-field-initializers
+      -Wnon-virtual-dtor
+      -Wchar-subscripts
+      -Wpointer-arith
+      -Woverloaded-virtual
+      -Wformat
+      -Wformat-security
+      -Werror=format-security
+      -Wuseless-cast
+      -Wno-c++11-extensions
+      -Wno-zero-length-array
+      -Wno-unused-parameter
+      -Wno-gnu-case-range
+      -Weffc++
+      -fpermissive
+      -fstack-protector-all
+      -fdata-sections
+      -ffunction-sections
+      -fvisibility=hidden
+      -fvisibility-inlines-hidden
+      -fno-limit-debug-info
+      -pipe
+      -pedantic
+      -stdlib=libc++
+    )
+
+    target_link_options(cxx_settings INTERFACE
+      -stdlib=libc++
+    )
+
+    if(DEFINED PLATFORM_MACOS)
+      target_compile_options(cxx_settings INTERFACE
+        -x objective-c++
+        -fobjc-arc
+        -Wabi-tag
+      )
+
+      target_link_options(cxx_settings INTERFACE
+        "SHELL:-framework AppKit"
+        "SHELL:-framework Foundation"
+        "SHELL:-framework CoreServices"
+        "SHELL:-framework CoreFoundation"
+        "SHELL:-framework CoreWLAN"
+        "SHELL:-framework CoreGraphics"
+        "SHELL:-framework DiskArbitration"
+        "SHELL:-framework IOKit"
+        "SHELL:-framework OpenDirectory"
+        "SHELL:-framework Security"
+        "SHELL:-framework ServiceManagement"
+        "SHELL:-framework SystemConfiguration"
+      )
+
+      target_link_libraries(cxx_settings INTERFACE
+        iconv
+        cups
+        bsm
+        xar
+      )
+    endif()
+
+    target_link_libraries(cxx_settings INTERFACE c++ c++abi)
+  endif()
+
+  target_compile_features(cxx_settings INTERFACE cxx_std_14)
+
+  # C settings
+  if (DEFINED PLATFORM_WINDOWS)
+    target_compile_options(c_settings INTERFACE
+      /std:c11
+      /MT
+      /EHs
+      /W3
+      /guard:cf
+      /bigobj
+    )
+  else()
+    target_compile_options(c_settings INTERFACE
+      -std=gnu11
+      -Qunused-arguments
+      -Wno-shadow-field
+      -Wall
+      -Wextra
+      -Wno-unused-local-typedef
```

Comment:
Is it possible to condense the similar C/C++ options?

Reply:
Yes, definitely.